events/janus_gelfevh.c

```diff
@@ -0,0 +1,476 @@
+	JANUS_VALIDATE_JSON_OBJECT(request, request_parameters,
+		error_code, error_cause, TRUE,
+		JANUS_GELFEVH_ERROR_MISSING_ELEMENT, JANUS_GELFEVH_ERROR_INVALID_ELEMENT);
+	if(error_code != 0)
+		goto plugin_response;
+	/* Get the request */
+	const char *request_text = json_string_value(json_object_get(request, "request"));
+	if(!strcasecmp(request_text, "tweak")) {
+		/* We only support a request to tweak the current settings */
+		JANUS_VALIDATE_JSON_OBJECT(request, tweak_parameters,
+			error_code, error_cause, TRUE,
+			JANUS_GELFEVH_ERROR_MISSING_ELEMENT, JANUS_GELFEVH_ERROR_INVALID_ELEMENT);
+		if(error_code != 0)
+			goto plugin_response;
+		/* Parameters we can change */
+		const char *req_events = NULL, *req_backend = NULL, *req_port = NULL;
+		/* Events */
+		if(json_object_get(request, "events"))
+			req_events = json_string_value(json_object_get(request, "events"));
+		/* Backend stuff */
+		if(json_object_get(request, "backend"))
+			req_backend = json_string_value(json_object_get(request, "backend"));
+		if(json_object_get(request, "port"))
+			req_port = json_string_value(json_object_get(request, "port"));
+		if(req_backend && req_port) {
+			/* Invalid backend address, port */
+			error_code = JANUS_GELFEVH_ERROR_INVALID_ELEMENT;
+			g_snprintf(error_cause, sizeof(error_cause), "Invalid backend URI '%s:%s'", req_backend, req_port);
+			goto plugin_response;
+		}
+		/* If we got here, we can enforce */
+		if(req_events)
+			janus_events_edit_events_mask(req_events, &janus_gelfevh.events_mask);
+		if(req_backend) {
+			janus_mutex_lock(&evh_mutex);
+			if(req_backend && req_port) {
+				g_free(backend);
+				g_free(port);
+				backend = g_strdup(req_backend);
+				port = g_strdup(req_port);
+			}
+			janus_mutex_unlock(&evh_mutex);
+		}
+	} else {
+		JANUS_LOG(LOG_VERB, "Unknown request '%s'\n", request_text);
+		error_code = JANUS_GELFEVH_ERROR_INVALID_REQUEST;
+		g_snprintf(error_cause, 512, "Unknown request '%s'", request_text);
+	}
+
+plugin_response:
+		{
+			json_t *response = json_object();
+			if(error_code == 0) {
+				/* Return a success */
+				json_object_set_new(response, "result", json_integer(200));
+			} else {
+				/* Prepare JSON error event */
+				json_object_set_new(response, "error_code", json_integer(error_code));
+				json_object_set_new(response, "error", json_string(error_cause));
+			}
+			return response;
+		}
+}
+
+/* Thread to handle incoming events */
+static void *janus_gelfevh_handler(void *data) {
+	JANUS_LOG(LOG_VERB, "Joining GelfEventHandler handler thread\n");
+	json_t *event = NULL, *output = NULL;
+	static char *event_text = NULL;
+	const char *short_message = NULL;
+
+	while(g_atomic_int_get(&initialized) && !g_atomic_int_get(&stopping)) {
+		event = g_async_queue_pop(events);
+		if(event == NULL)
+			continue;
+		if(event == &exit_event)
+			break;
+		output = NULL;
+
+		while(TRUE) {
+			/* Handle event */
+			int type = json_integer_value(json_object_get(event, "type"));
+			switch(type) {
```

Comment:
There's no need for all that: you can just use `janus_events_type_to_label` or `janus_events_type_to_name` from `events.h`(unless they really need to be in the format you're using here?)

Reply:
this saves me a lot of code :D